nautobot/extras/fields.py

```diff
@@ -0,0 +1,96 @@
+from functools import partialmethod
+
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+from django.utils.encoding import force_str
+from django.utils.hashable import make_hashable
+
+from nautobot.utilities.forms import DynamicModelChoiceField
+
+
+class LimitedChoiceField(models.ForeignKey):
```

Comment:
Can we give this a less generic name, perhaps something like `ForeignKeyLimitedByContentTypes` or `ContentTypeLimitedForeignKey`?